caput/memh5.py

```diff
@@ -2330,21 +2602,34 @@ def _distributed_group_to_hdf5_serial(
                 mode,
                 convert_dataset_strings=convert_dataset_strings,
                 convert_attribute_strings=convert_attribute_strings,
-                **kwargs
+                **kwargs,
             )
 
         # Write out distributed datasets (only the data, the attributes are written below)
         elif isinstance(entry, MemDatasetDistributed):
 
             arr = check_unicode(entry)
 
-            arr.to_hdf5(
-                fname,
-                entry.name,
-                chunks=entry.chunks,
-                compression=entry.compression,
-                compression_opts=entry.compression_opts,
-            )
+            if fileformats.HDF5.compression_enabled():
+                arr.to_hdf5(
+                    fname,
+                    entry.name,
+                    chunks=entry.chunks,
+                    *fileformats.HDF5.compression_kwargs(
+                        compression=entry.compression,
+                        compression_opts=entry.compression_opts,
+                    ),
+                )
+            else:
+                # disable compression if not enabled for HDF5 files
+                # https://github.com/chime-experiment/Pipeline/issues/33
+                arr.to_hdf5(
+                    fname,
+                    entry.name,
+                    chunks=None,
+                    compression=None,
+                    compression_opts=None,
+                )
```

Comment:
It seems like this would be better structured with the conditional formatting the arguments and then a single call to `.to_hdf5`

Reply:
Or move the conditional into `.to_hdf5`